pkg/utils/apply/apply.go

```diff
@@ -170,6 +170,16 @@ func (a *APIApplicator) Apply(ctx context.Context, desired client.Object, ao ...
 		return nil
 	}
 
+	if utilfeature.DefaultMutableFeatureGate.Enabled(features.ApplyResourceByUpdate) && isUpdatableResource(desired) {
```

Comment:
I will use switch in this code.

switch{
case utilfeature.DefaultMutableFeatureGate.Enabled(features.ApplyResourceByUpdate) && isUpdatableResource(desired):
default:
}

Reply:
But I think there will be no other cases here, even in the future. Is there any plan for more strategies other than Update&Patch?

Reply:
A switch case is used in isUpdatableResource.

Reply:
A switch will be more clear even there's only two cases

Reply:
OK, agree.

Reply:
Done